src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsJavaDocsTest.java

```diff
@@ -393,35 +399,63 @@ private static String emptyStringArrayDefaultValue(Node defaultValueNode,
 
-        for (Node child = node.getFirstChild(); child != null; child = child.getNextSibling()) {
-            if (child.getNodeType() == Node.TEXT_NODE) {
-                for (String temp : child.getTextContent().split("\n")) {
-                    final String text = temp.trim();
+        boolean skipExampleSection = false;
+        boolean alreadySkippedExamplesSection = false;
 
-                    if (!text.isEmpty()) {
-                        if (shouldAppendSpace(result, text.charAt(0))) {
-                            result.append(' ');
-                        }
-
-                        result.append(text);
-                    }
+        for (Node child = node.getFirstChild(); child != null; child = child.getNextSibling()) {
+            if (MODULES_EXAMPLES_TO_SKIP.contains(checkName)) {
+                // Stop skipping examples section when we encounter parent section
+                if ("p".equals(child.getNodeName())
+                        && child.getTextContent().startsWith("\nParent is")) {
+                    skipExampleSection = false;
+                    alreadySkippedExamplesSection = true;
                 }
-            }
-            else {
-                if (child.hasAttributes() && child.getAttributes().getNamedItem("class") != null
-                        && "wrapper".equals(child.getAttributes().getNamedItem("class")
-                        .getNodeValue())) {
-                    appendNodeText(result, XmlUtil.getFirstChildElement(child));
+                if (skipExampleSection && !alreadySkippedExamplesSection) {
+                    continue;
                 }
-                else {
-                    appendNodeText(result, child);
+                // Examples section is located after property section
+                if ("ul".equals(child.getNodeName())
+                        && child.getTextContent().startsWith("\n\nProperty")) {
+                    skipExampleSection = true;
                 }
             }
+
+            if (child.getNodeType() == Node.TEXT_NODE) {
+                handleTextNode(result, child);
+            }
+            else {
+                handleNonTextNode(result, child);
+            }
         }
 
         return result.toString();
     }
 
+    private static void handleTextNode(StringBuilder result, Node child) {
+        for (String temp : child.getTextContent().split("\n")) {
+            final String text = temp.trim();
+
+            if (!text.isEmpty()) {
+                if (shouldAppendSpace(result, text.charAt(0))) {
+                    result.append(' ');
+                }
+
+                result.append(text);
+            }
+        }
+    }
+
+    private static void handleNonTextNode(StringBuilder result, Node child) {
+        if (child.hasAttributes() && child.getAttributes().getNamedItem("class") != null
+                && "wrapper".equals(child.getAttributes().getNamedItem("class")
+                .getNodeValue())) {
```

Comment:
Hard to read.
Please make variable for `child.getAttributes().getNamedItem("class")` and yes there will be one more nesting scope

Reply:
Done.